plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -1672,12 +1689,124 @@ def update_realm_role(self, rolerep, realm='master'):
         """
         role_url = URL_REALM_ROLE.format(url=self.baseurl, realm=realm, name=quote(rolerep['name']))
         try:
-            return open_url(role_url, method='PUT', http_agent=self.http_agent, headers=self.restheaders, timeout=self.connection_timeout,
-                            data=json.dumps(rolerep), validate_certs=self.validate_certs)
+            composites = None
+            if "composites" in rolerep:
+                composites = copy.deepcopy(rolerep["composites"])
+                del rolerep["composites"]
+            role_response = open_url(role_url, method='PUT', http_agent=self.http_agent, headers=self.restheaders, timeout=self.connection_timeout,
+                                     data=json.dumps(rolerep), validate_certs=self.validate_certs)
+            self.update_role_composites(rolerep=rolerep, composites=composites, realm=realm)
```

Comment:
... and instead check it here:
```suggestion
            if composites is not None:
                self.update_role_composites(rolerep=rolerep, composites=composites, realm=realm)
```
That makes it more obvious that this code now only behaves differently if `composites` is present.